cmd/chloggen/entry.go

```diff
@@ -0,0 +1,71 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"strings"
+)
+
+type Entry struct {
+	ChangeType string `yaml:"change_type"`
+	Component  string `yaml:"component"`
+	Note       string `yaml:"note"`
+	Issues     []int  `yaml:"issues"`
+}
+
+var changeTypes = []string{
+	breaking,
+	deprecation,
+	newComponent,
+	enhancement,
+	bugFix,
+}
+
+func (e Entry) Validate() error {
+	var validType bool
+	for _, ct := range changeTypes {
+		if e.ChangeType == ct {
+			validType = true
+			break
+		}
+	}
+	if !validType {
+		return fmt.Errorf("'%s' is not a valid 'change_type'. Specify one of %v", e.ChangeType, changeTypes)
+	}
+
+	if e.Component == "" {
+		return fmt.Errorf("specify a 'component'")
+	}
+
+	if e.Note == "" {
+		return fmt.Errorf("specify a 'note'")
+	}
+
+	if len(e.Issues) == 0 {
```

Comment:
If an associated issue is present but not a valid issue number then check-links will fail.  How should we validate that the supplied issue number is legit?

Reply:
Ok, we can manually check the issue in the new file is the same issue linked in the PR